### PR TITLE
Fixed issues #6 and #7

### PR DESCRIPTION
**Pull request by edegaudenzi**
Fixed issues #6 and #7: escaped filename and dynamically assigned the temp path.

### CODE REVIEW

**Comment by xthiago** [comment]
Hi @edegaudenzi. Nice catch! I really appreciate your contribution.

- The unit tests executed by travis was failed. There is an error on line 26. See the output here: https://travis-ci.org/xthiago/pdf-version-converter/jobs/189564296

- Would be nice if our unit tests check filenames with special characters in order to avoid regressions with this problem. What you think about adding some files to `GhostscriptConverterCommandTest::filesProvider`?

Could you do these adjustments?

**Comment by edegaudenzi** [comment]
Hi man, it is a pleasure!

about the first error, I should check it deeper because it is a native 
php instruction, I don t know why is failing tests. About the second 
one, yes I thought to populate with some new unit tests. I will do that 
asap.

**Comment by xthiago** [comment]
The error happens because you are calling the function `sys_get_temp_dir()` on the declaration of `$tmp` property.

The PHP Manual says (http://php.net/language.oop5.properties):

>[..] This declaration may include an initialization, but this initialization must be a constant value--that is, it must be able to be evaluated at compile time and must not depend on run-time information in order to be evaluated.

In order to fix that you need to move that call to `__construct()` function.

**Comment by edegaudenzi** [comment]
Yeah, I copied and pasted in the wrong part of the code, sorry. In fixing it, I also changed the if condition with a ternary (full ternary to be compliant with 5.2.1), code seems tidier.

File: tests/Converter/GhostscriptConverterCommandTest.php [hunk 1]
```diff
@@ -69,6 +70,8 @@ public function testMustConvertPDFVersionWithSuccess($file, $newVersion)
     {
         $tmpFile = $this->tmp .'/'. uniqid('pdf_version_changer_test_') . '.pdf';
 
+        echo "$tmpFile \n";
```
Comment (xthiago) [inline tests/Converter/GhostscriptConverterCommandTest.php]: Please remove this line.

File: tests/Converter/GhostscriptConverterCommandTest.php [hunk 2]
```diff
@@ -32,6 +32,7 @@ class GhostscriptConverterCommandTest extends PHPUnit_Framework_TestCase
         'v1.6.pdf',
         'v1.7.pdf',
         'v2.0.pdf',
+        'v2.0 filename with "Sp3ci4l"; <\'Ch4r5\'> !£$%&()=?^[]{}è@#§.pdf',
```
Comment (xthiago) [inline tests/Converter/GhostscriptConverterCommandTest.php:35]: Wow! It's a good name :)

File: src/Converter/GhostscriptConverter.php [hunk 3]
```diff
@@ -44,10 +44,7 @@ public function __construct(GhostscriptConverterCommand $command, Filesystem $fs
     {
         $this->command = $command;
         $this->fs = $fs;
-
-        if ($tmp) {
-            $this->tmp = $tmp;
-        }
+        $this->tmp = empty($tmp)?sys_get_temp_dir():$tmp;
```
Comment (xthiago) [inline src/Converter/GhostscriptConverter.php]: Looks good. Could you add space between the operators `?` and `:`?

**Review by xthiago — COMMENTED** [review]
Thank you for the adjustments. Could you do some minor fixes?

File: src/Converter/GhostscriptConverter.php (hunk 3 above)
Comment (edegaudenzi) [reply src/Converter/GhostscriptConverter.php]: if you don't care about 5.2.1 support (travis is starting from 5.3) I can also write the shorthand for that

File: tests/Converter/GhostscriptConverterCommandTest.php (hunk 1 above)
Comment (edegaudenzi) [reply tests/Converter/GhostscriptConverterCommandTest.php]: Ooopss.. :)

File: src/Converter/GhostscriptConverter.php (hunk 3 above)
Comment (xthiago) [reply src/Converter/GhostscriptConverter.php]: Our dependencies (from Symfony) requires PHP 5.3. Also, PHP 5.2 is too old. Are you using it?

What you thing about: `$this->tmp = empty($tmp) ? sys_get_temp_dir() : $tmp;`?

Comment (edegaudenzi) [reply src/Converter/GhostscriptConverter.php]: If you need 5.3 you can use:
$this->tmp = $tmp ? : sys_get_temp_dir();

It's cleaner, isn't it?

Comment (xthiago) [reply src/Converter/GhostscriptConverter.php]: go ahead 👍

**Comment by xthiago** [comment]
The merge was done. Thank you @edegaudenzi!

**Pull request merged**